packages/dds/tree/src/simple-tree/treeApi.ts

```diff
@@ -140,11 +155,61 @@ export const treeNodeApi: TreeNodeApi = {
 
 /**
  * A collection of events that can be raised by a {@link TreeNode}.
+ *
+ * @privateRemarks
+ * TODO: add a way to subscribe to a specific field (for afterShallowChange and afterDeepChange).
+ * Probably have object node and map node specific APIs for this.
+ *
+ * TODO: ensure that subscription API for fields aligns with API for subscribing to the root.
+ *
+ * TODO: add more wider area (avoid needing tons of afterShallowChange registration) events for use-cases other than afterDeepChange.
+ * Some ideas:
+ *
+ * - afterDeepChange, but with some subtrees/fields/paths excluded
+ * - helper to batch several afterShallowChange calls to a afterDeepChange scope
+ * - parent change (ex: registration on the parent field for a specific index: maybe allow it for a range. Ex: node event takes optional field and optional index range?)
+ * - new content inserted into subtree. Either provide event for this and/or enough info to afterDeepChange to find and search the new sub-trees.
+ * Add separate (non event related) API to efficiently scan tree for given set of types (using low level cursor and schema based filtering)
+ * to allow efficiently searching for new content (and initial content) of a given type.
+ *
  * @public
  */
-export interface TreeNodeEvents {
+export interface TreeChangeEvents {
+	/**
+	 * Raised on a node after a change is applied to one of its fields.
```

Comment:
Do we surface the concept of a "field" elsewhere in this API? It would probably be useful to point to a definition, or use different terminology.

Reply:
Good callout... @CraigMacomber @taylorsw04 any thoughts? I don't see a lot of obviously-exposed, field-related APIs/information on simple tree. The one instance I found was in the docs for `TreeNodeApi.key()` ("this returns the key of the _field_ that it is under "):

```
	/**
	 * The key of the given node under its parent.
	 * @remarks
	 * If `node` is an element in a {@link (TreeArrayNode:interface)}, this returns the index of `node` in the array node (a `number`).
	 * Otherwise, this returns the key of the field that it is under (a `string`).
	 */
	key(node: TreeNode): string | number;
```

Should fields be something that the users of simple-tree are aware of? Or would it be better to write docs in terms of a node's _properties_ (happy to hear other ideas) instead?